### ISSUE TITLE
warning: used import from `std` instead of `core` in external macros

### ISSUE THREAD

**Issue by CBenoit** [rust-lang/rust-clippy#11970]
### Summary

When enabling `std_instead_of_core`,  the warning is emitted for code derived using derive macros.

This is likely not the desired behavior since there is a check using `in_external_macro` in the implementation.

### Lint Name

std_instead_of_core

### Reproducer

I tried this code:

```rust
#![warn(clippy::std_instead_of_core)]

use arbitrary::Arbitrary;

#[derive(Arbitrary)]
pub struct SomeStruct;
```

I saw this happen:

```
warning: used import from `std` instead of `core`
  --> crates/ironvnc-core/src/pixel_format.rs:10:42
   |
10 | #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
   |                                          ^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider importing the item from `core`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_core
   = note: this warning originates in the derive macro `arbitrary::Arbitrary` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I expected to see this happen: no warning

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```
